semanticdb-javac/src/main/java/com/sourcegraph/semanticdb_javac/RangeFinder.java

```diff
@@ -30,6 +30,9 @@
     int endPos = (int) trees.getSourcePositions().getEndPosition(root, path.getLeaf());
     // false for anonymous classes
     if (name.length() != 0) {
+      // for declared constructors with no modifiers, findNameIn will return -1 as the name will
+      // not start with a space, the first char is the start of the name
+      if (source.charAt(startPos) != ' ') startPos -= 1;
```

Comment:
How do we know `startPos -= 1` fixes this problem?

Reply:
Youre right, that is an assumption, with holes. Its currently based on the assumption that
1. the file uses spaces for indentation
2. the constructor is indented aka is preceded by spaces
3. any type parameter bounds do not have the constructor name as a prefix aka `<T extends Stringer> String()`

An example of where point 3 falls apart can be seen here:
![image](https://user-images.githubusercontent.com/18282288/114730428-c42dd700-9d38-11eb-8ace-536454d2e5d6.png)

Reply:
All is well, constructor preferred position starts at the name of the constructor : ) :tada: 

![image](https://user-images.githubusercontent.com/18282288/114734818-7a46f000-9d3c-11eb-9215-182acd71a423.png)